### PR TITLE
Improve logging of goal state processing

### PR DESCRIPTION
**Pull request by narrieta**
Improved the log to distinguish between fast track and fabric goal states.

The current log is based on incarnation numbers; with this change we use the incarnation/etag as appropriate and we also add the source (FastTrack/Fabric) of the goal state and the channel (WireServer/HostGAPlugin) used to fetch it.

Some sample messages:

```
2022-02-27T17:51:55.239526Z INFO ExtHandler ExtHandler Fetching goal state [incarnation 4]
2022-02-27T17:51:55.281315Z INFO ExtHandler ExtHandler Fetched new vmSettings [correlation ID: None eTag: 2824367392948713696 source: Fabric]
2022-02-27T17:51:55.319733Z INFO ExtHandler ExtHandler Certificate with thumbprint 4D0D33037EE6205C5BB0868F097F274B1192070D has no matching private key.
2022-02-27T17:51:55.320105Z INFO ExtHandler ExtHandler Fetch goal state completed
2022-02-27T17:51:55.320657Z INFO ExtHandler ExtHandler ProcessExtensionsGoalState started [etag_2824367392948713696 channel: HostGAPlugin source: Fabric activity: f7d4adf8-2b0a-422e-a58e-cd2f672697f2 correlation 815ff807-ebd1-41ae-ae57-a9619bcffada created: 2022-02-27T17:51:45.906754Z]
...
2022-02-27T17:51:59.394225Z INFO ExtHandler ExtHandler ProcessExtensionsGoalState completed [etag_2824367392948713696 4073 ms]
...
2022-02-27T17:54:46.845170Z INFO ExtHandler ExtHandler Fetched new vmSettings [correlation ID: None eTag: 12730129496418900798 source: FastTrack]
2022-02-27T17:54:46.845836Z INFO ExtHandler ExtHandler ProcessExtensionsGoalState started [etag_12730129496418900798 channel: HostGAPlugin source: FastTrack activity: 512e2839-45b6-4ccd-9324-b877c020e652 correlation 1f49b21b-5f08-446e-a254-adb4c5c59d99 created: 2022-02-27T17:54:43.640938Z]
...
2022-02-27T17:54:50.932657Z INFO ExtHandler ExtHandler ProcessExtensionsGoalState completed [etag_12730129496418900798 4086 ms]

```

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -294,53 +293,43 @@ def __last_gs_unsupported(self):
...
-            extensions_goal_state = self.protocol.get_goal_state().extensions_goal_state
+            gs = self.protocol.get_goal_state()
+            egs = gs.extensions_goal_state
 
-            # self.ext_handlers and etag need to be initialized first, since status reporting depends on them; also
+            # self.ext_handlers needs to be initialized before returning, since status reporting depends on it; also
             # we make a deep copy of the extensions, since changes are made to self.ext_handlers while processing the extensions
-            self.ext_handlers = copy.deepcopy(extensions_goal_state.extensions)
-            etag = self.protocol.client.get_goal_state().incarnation
+            self.ext_handlers = copy.deepcopy(egs.extensions)
```
Comment (kevinclark19a) [inline azurelinuxagent/ga/exthandlers.py:302]: Would it make sense to move this copying into the `self.protocol.get_goal_state()` call? We're concerned about changing the goal state object whilst processing the `ext_handlers`, but we're still leaving a mutable reference to the goal state object itself above (`gs` and `egs`).

File: azurelinuxagent/ga/exthandlers.py [hunk 2]
```diff
@@ -294,53 +293,43 @@ def __last_gs_unsupported(self):
...
+                duration = elapsed_milliseconds(utc_start)
+                if error is None:
+                    message = 'ProcessExtensionsGoalState completed [{0} {1} ms]'.format(egs.id, duration)
+                    logger.info(message)
+                else:
+                    message = 'ProcessExtensionsGoalState failed [{0} {1} ms]\n{2}'.format(egs.id, duration, error)
+                    logger.error(message)
+                add_event(op=WALAEventOperation.ExtensionProcessing, is_success=(error is None), message=message, log_event=False, duration=duration)
+
         except Exception as error:
```
Comment (kevinclark19a) [inline azurelinuxagent/ga/exthandlers.py:329]: Why move this down and nest the above try except? I'm not quite seeing what this covers over where it was before.

File: azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py [hunk 3]
```diff
@@ -74,28 +75,37 @@ def schema_version(self):
...
+        """
+        The correlation id for the CRP operation
+        """
         return self._correlation_id
 
+    @property
+    def fetch_correlation_id(self):
+        """
+        The correlation id for the fetch operation (i.e. the call to the HostGAPlugin vmSettings API)
+        """
```
[the remark this answers is not in the document]
Comment (narrieta) [reply azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py:94]: thanks; fixed

File: azurelinuxagent/ga/exthandlers.py (hunk 2 above)
Comment (narrieta) [reply azurelinuxagent/ga/exthandlers.py:329]: i find a single try/except more readable than 2 separate blocks. the new implementation preserves the previous behavior

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (kevinclark19a) [reply azurelinuxagent/ga/exthandlers.py:302]: Can we add a `# TODO` to capture this in the code?

**Review by nagworld9 — APPROVED** [review]

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (narrieta) [reply azurelinuxagent/ga/exthandlers.py:302]: it's a big TODO: v3 :)

**Review by kevinclark19a — APPROVED** [review]

**Pull request merged**